.github/workflows/publish-release.yml

```diff
@@ -3,22 +3,227 @@ on:
+        cp -Lr ../../bin ../../example ../../lib ../../test realm_dart
+        rsync -vt ../../* realm_dart/
+        cd realm_dart
+        dart pub publish --dry-run || true
+      working-directory: release/${{ steps.get-version.outputs.version }}
+
+    - name: Extract Changelog
+      run: |
+        $match = Get-Content ../../CHANGELOG.md -Raw | select-string -pattern "(?ms)^(?<latestVersionChanges>.+?)(?=(\n|\r\n)## )"
+        $latestChanges = $match.Matches[0].Groups["latestVersionChanges"].Value;
+        $latestChanges | Out-File ExtractedChangelog.md
+      working-directory: release/${{ steps.get-version.outputs.version }}
+      shell: pwsh
+
+    - name: Upload release folder
+      uses: actions/upload-artifact@v2
+      with:
+        name: release-bundle
+        path: release/**
+        retention-days: 30
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_S3_ACCESS_KEY }}
+        aws-secret-access-key: ${{ secrets.AWS_S3_SECRET_KEY }}
+        aws-region: us-east-1
+
+    - name: Upload release folder to S3
+      id: upload-to-s3
+      run: |
+        tar -zcvf packages.tar.gz common generator realm_dart realm ExtractedChangelog.md
+        rm -rf common generator realm_dart realm ExtractedChangelog.md
+
+        s3_folder="static.realm.io/downloads/dart/${{ steps.get-version.outputs.version }}"
+        aws s3 sync --acl public-read . "s3://$s3_folder"
+        echo "::set-output name=url::https://$s3_folder/packages.tar.gz"
+      working-directory: release/${{ steps.get-version.outputs.version }}
+
+  publish-packages:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.inputs.environment == 'Production' }}
+    environment:
+      name: 'Production'
+      url: https://pub.dev/packages/realm/versions/${{ needs.prepare-packages.outputs.version }}
+    name: Publish release
+    needs:
+    - prepare-packages
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+        submodules: false
+
+    - name: Download release folder
+      uses: actions/download-artifact@v2
+      with:
+        name: release-bundle
+        path: release
+
+    - name : Setup Dart SDK
+      uses: dart-lang/setup-dart@main
+      with:
+        sdk: stable
+
+    # TODO: remove --dry-run
+    - name: Publish packages to pub.dev
+      run: |
+        dart pub publish --dry-run --directory common || true
+        dart pub publish --dry-run --directory generator || true
+        dart pub publish --dry-run --directory realm_dart || true
+        dart pub publish --dry-run --directory realm || true
+      working-directory: release/${{ needs.prepare-packages.outputs.version }}
+
+    - name: Find Release PR
+      uses: juliangruber/find-pull-request-action@f9f7484f8237cf8485e5ab826e542ba5dd9e9c6e
+      id: find-pull-request
+      with:
+        branch: ${{ github.ref }}
+
+    - name: Merge Pull Request
```

Comment:
Should these steps be checking a success status of something before it. Unsure what will happen if we fail to upload binaries to S3 and we merge to master this the release PR

Reply:
The job fails it any of its steps fails, so we won't get this far if something earlier fails.